### PR TITLE
Opencore 0.6.5 upgrade with latest Clover integration

### PR DESCRIPTION
**Pull request by serdeliuk**
@jief666 this upgrade seems to be mandatory in order to be able to boot newer oses with SecureBootModel enabled

### CODE REVIEW

**Comment by jief666** [comment]
Could you have a look, it's failing.
Looks like just an unused var.

**Comment by serdeliuk** [comment]
I have commented-out the unused variable

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,4 @@
 [submodule "OpenCorePkg"]
+	url = https://github.com/serdeliuk/OpenCorePkg.git
```
Comment (stevezhengshiqi) [inline .gitmodules:2]: Hi, are you sure to use your fork of OC instead of CloverHackyColor's?

Comment (serdeliuk) [reply .gitmodules:2]: I have modified my OC indeed, https://github.com/serdeliuk/OpenCorePkg/commit/07db7299071b2391f4644fe81a35635b2fd70897
then requested another pull for OC

Comment (jief666) [reply .gitmodules:2]: Well spotted. I would have done the same mistake. He has to fork to create the pull request.
Once OpenCore is fixed and the PR merged, he can switch back our fork. Then, we merge PR for Clover itself.

Comment (jief666) [reply .gitmodules:2]: @serdeliuk OpenCorePKG PR merged. Could you now switch back to CloverHackyColor's fork for OpenCorePKG. Then commit that change.

Comment (serdeliuk) [reply .gitmodules]: done

Comment (serdeliuk) [reply .gitmodules]: This workflow build has a cache? Can be cleaned or completely deleted?

Comment (stevezhengshiqi) [reply .gitmodules]: You have to `git submodule update --remote` to pull the latest OC branch.

Comment (serdeliuk) [reply .gitmodules]: Nice catch, thanks, done.

Comment (serdeliuk) [reply .gitmodules]: Haha, nice, i didn't expected that an unused variable error actually to be a false positive and didn't checked more, if i try to compile the code with that variable commented-out i have the  same error now.
Could be something related to the building environment? I use  Mojave to build Clover.

Comment (serdeliuk) [reply .gitmodules]: Is there any way to replicate the build env on my computer?

Comment (serdeliuk) [reply .gitmodules]: So, there is something wrong with the testing env, the Privilege variable is set and used, i wonder why the unused error was in the first place. I have fixed back the OC and requested another pull

Comment (serdeliuk) [reply .gitmodules]: It seems that the uploaded OpenCore.c has some conditionals missing, probably i applied a wrong patch, i am fixing the OC and request another pull

**Comment by jief666** [comment]
@serdeliuk maybe you were a bit too fast on that one :-).
If you comment out a var, you also have to comment out assigment

> Privilege = &mOpenCorePrivilege;

NOTE : you second commit "Changed OpenCorePkg module URL" has a wrong label.

File: .gitmodules (hunk 1 above)
Comment (jief666) [reply .gitmodules]: I use Mojave too.
You can build with buildme script.

Comment (serdeliuk) [reply .gitmodules]: Yes, i can, but the dev testing workflow in the beginning throwed an  error that i didn't had, probably will be better to understand how the workflow works and what env use, at least i can test locally prior uploading, this way i can save everyone's time :)
I found the issue, my last patch had some missing  compile time conditions , i will fix my OC and issue later today another pull request, then see

**Comment by jief666** [comment]
And if you put ".gitmodules" into your .git/info/exclude, you'll be able to switch to your branch without committing the wrong .gitmodules.

**Comment by jief666** [comment]
OpenCore is merged. You now have to commit in this branch the commit reference of OpenCorePkg, I think. Which should trigger the automatic build.
I imagine you have to switch back to CloverHackyColor's version of OpenCorePkg.

**Comment by serdeliuk** [comment]
> OpenCore is merged. You now have to commit in this branch the commit reference of OpenCorePkg, I think. Which should trigger the automatic build.
> I imagine you have to switch back to CloverHackyColor's version of OpenCorePkg.

Yup, changed the OC url to Clover's repo

File: .gitmodules (hunk 1 above)
Comment (serdeliuk) [reply .gitmodules]: > Try a build with "buildme" script.
Yes, this did the trick.
I wonder if anyone else  observed  that buildme script build a smaller efi than ebuild, is there any obvious reason why is doing this? 
```
-rw-r--r--   1 marc  admin   1.4M Sep 15 16:09 CLOVERX64.efi_buildme
-rw-r--r--   1 marc  admin   1.6M Sep 15 16:12 CLOVERX64.efi_ebuild

```

Comment (jief666) [reply .gitmodules]: buildme build clover 2 times, with different options. buildme always build DUET. So I wouldn't be surprised there is room for optimization...

**Comment by jief666** [comment]
@serdeliuk I've just came across some lost modification after 0.6.5 update. For example, line OpenCoreMisc.c:552

>   DEBUG ((DEBUG_INFO, "OC: Watchdog status is %d\n", Config->Misc.Debug.DisableWatchDog == FALSE));

was enclosed in a #ifdef

This is not here anymore !
Did you run through all our modifications ?

**Comment by jief666** [comment]
OpenCorePkg.dsc:171 also was lost in the update.
I'm starting to think that we will have to revert, because there maybe be a lot of lost modifications...

**Comment by serdeliuk** [comment]
@jief666 what is strage is that i have the lines in the initial patch, probably i applied a wrong patch to OpenCoreMisc.c too. i wil review all of them again.

what was at OpenCorePkg.dsc:171 ?

**Comment by jief666** [comment]
It was commented out so I can still compile a stand alone OpenCore.

**Comment by jief666** [comment]
Hi.
I've decided to integrate the lastest OC, reorganize OpenCorePkg commits in a way so I can cherry pick commits for the next version.
Does latest version of OC works for you ?
If yes, you can just wait instead of going over the patches.

**Comment by serdeliuk** [comment]
For me Clover built with OC 0.6.5 works well, i have tested with misc enabled too and config-oc.plist (provided)
The default settings were not changed in the pull request, i have a personal build for that.

**Comment by jief666** [comment]
It's ok. I'll do a branch.
But not knowing if all the OC modification are imported is a bit annoying as it's impossible to remember all what they solve. So maybe you didn't imported few things that doesn't apply to you. But that's still regression because I don't modify for the sake of it. Even if it's just debug message from OC, even if they are commented, they are usually useful.

Doing it automatically without manual verification usually doesn't work well. Speak from experience.

**Comment by serdeliuk** [comment]
Just wait a day or so, I will have another look on patches and what id did that they didn't applied, or you can switch back Clover's module to last known good OC branch until i dig deeper?

**Comment by serdeliuk** [comment]
For sure i overlooked a failed hunk that passed not applied

**Comment by jief666** [comment]
@serdeliuk I hope you'll have a chance to try the 0.7.3 version...

**Comment by serdeliuk** [comment]
> @serdeliuk I hope you'll have a chance to try the 0.7.3 version...

yup i am going to upgrade, but starting with 0.6.6 they removed and/or renamed a lot of functions/libraries, for example there is no Platform/OpenCore anymore, they changed a lot in the structure...

**Comment by jief666** [comment]
I meant : I did the upgrade. It's already done in a git branch.
NOTE : Platform/OpenCore went to Application/OpenCore
So if you can just try that it works for you, that'd be great.

**Comment by serdeliuk** [comment]
Aaa..... niceee, let give a try, i will let you know the result

**Pull request merged**